### PR TITLE
use engine-api for events

### PR DESCRIPTION
**Pull request by vieux**
This one is too risky to merge for the rc3, let's wait until the next release.

### CODE REVIEW

**Comment by vieux** [comment]
ping @nishanttotla

**Comment by vieux** [comment]
WIP

**Comment by nishanttotla** [comment]
I agree we should keep this for after 1.2. Do you know if `engine-api` is willing to add the event package?

**Comment by vieux** [comment]
@nishanttotla I don't think so, but you can do it in swarm (as I did)

**Comment by vieux** [comment]
@nishanttotla @dongluochen please review this one. not sure why github says pending, all tests are finished and green.

File: cluster/event_monitor.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+	resultChan := make(chan decodingResult)
+
+	go func() {
+		dec := json.NewDecoder(responseBody)
+		for {
+			var result decodingResult
+			result.err = dec.Decode(&result.msg)
+			resultChan <- result
+			if result.err != nil && result.err == io.EOF {
```
Comment (dongluochen) [inline cluster/event_monitor.go]: `result.err != nil` is redundant here.

**Comment by dongluochen** [comment]
LGTM

**Comment by nishanttotla** [comment]
Could we make a separate package for events (in Swarm, if engine-api doesn't intend to do it right now)? Also perhaps we should wait for #2124?

**Comment by vieux** [comment]
@nishanttotla I think engine-api will do it

File: Godeps/Godeps.json [hunk 2]
```diff
@@ -129,6 +129,11 @@
 			"Rev": "5b9e86c11862e755be8871b64aada2a65ded4b8b"
 		},
 		{
+			"ImportPath": "github.com/docker/engine-api/types/events",
+			"Comment": "v0.3.1-16-g6ca9064",
+			"Rev": "6ca9064650fd3d66ea1722c24e475ec0138feb11"
```
Comment (nishanttotla) [inline Godeps/Godeps.json]: Why is this commit hash different from the other packages in `engine-api`?

Comment (vieux) [inline Godeps/Godeps.json]: probably mixed up during rebase. will fix

**Comment by vieux** [comment]
PTAL @nishanttotla

File: Godeps/Godeps.json (hunk 2 above)
Comment (vieux) [inline Godeps/Godeps.json]: fixed

File: cluster/engine_test.go [hunk 3]
```diff
@@ -25,6 +26,13 @@ import (
 	"github.com/stretchr/testify/mock"
 )
 
+type infinitRead struct{}
```
Comment (abronan) [inline cluster/engine_test.go]: @vieux nit: you meant `infiniteRead` I guess?

**Comment by abronan** [comment]
LGTM

@vieux can you push to trigger the tests with `1.11` as well? (after fixing the small typo)

**Comment by nishanttotla** [comment]
I'm confused why only the events package is updated in the `Godeps.json`, shouldn't all packages be at the same commit?

**Comment by nishanttotla** [comment]
LGTM 🎊

**Pull request merged**